catalyst-gateway/bin/src/service/api/documents/get_document.rs

```diff
@@ -27,17 +28,48 @@ pub(crate) enum Responses {
 pub(crate) type AllResponses = WithErrorResponses<Responses>;
 
 /// # GET `/document`
-#[allow(clippy::unused_async, clippy::no_effect_underscore_binding)]
 pub(crate) async fn endpoint(document_id: uuid::Uuid, version: Option<uuid::Uuid>) -> AllResponses {
+    match get_document(&document_id, version.as_ref()).await {
+        Ok(doc) => {
+            match minicbor::to_vec(&doc) {
```

Comment:
Don't use minicbor directly, functionality must be provided by signed docs library crate.